### PR TITLE
Vas 17

### PR DESCRIPTION
**Pull request by vguzmanp**

### CODE REVIEW

File: VAS.KPI/Services/StorageService.cs [hunk 1]
```diff
@@ -0,0 +1,353 @@
+﻿// <copyright file="Storage.cs" company="Microsoft">
+// Copyright © Microsoft. All Rights Reserved.
+// </copyright>
+
```
Comment (aameijide) [inline VAS.KPI/Services/StorageService.cs]: Incorrect copyright file

File: VAS.KPI/Services/StorageService.cs [hunk 2]
```diff
@@ -0,0 +1,353 @@
...
+	using System.Security.Cryptography;
+	using System.Security.Principal;
+	using System.Text;
+	using System.Threading;
+	using System.Threading.Tasks;
+	using Microsoft.HockeyApp.Channel;
+	using Microsoft.HockeyApp.Extensibility.Implementation;
+	using Microsoft.HockeyApp.Extensibility.Implementation.Tracing;
+	using Microsoft.HockeyApp.Services;
+	using Services;
```
Comment (aameijide) [inline VAS.KPI/Services/StorageService.cs]: usings outside the namespace

File: VAS.KPI/Services/StorageService.cs [hunk 3]
```diff
@@ -0,0 +1,353 @@
...
+			this.CapacityInBytes = 10 * 1024 * 1024; // 10 MB
+			this.MaxFiles = 5000;
+
+			//Task.Factory.StartNew (this.DeleteObsoleteFiles)
+			//	.ContinueWith (
+			//		task => {
+			//			string msg = string.Format (CultureInfo.InvariantCulture, "Storage: Unhandled exception in DeleteObsoleteFiles: {0}", task.Exception);
+			//			CoreEventSource.Log.LogVerbose (msg);
+			//		},
+			//		TaskContinuationOptions.OnlyOnFaulted);
```
Comment (aameijide) [inline VAS.KPI/Services/StorageService.cs]: Commented code

**Comment by aameijide** [comment]
Ouch, comments appears as outdated because I commented in the commit and it was moved later, but the issues are still there

**Comment by vguzmanp** [comment]
Okk

File: VAS.KPI/HockeyPlatformHelperMono.cs [hunk 4]
```diff
@@ -0,0 +1,320 @@
...
+
+		private const string FILE_PREFIX = "HA__SETTING_";
+		IsolatedStorageFile isoStore = IsolatedStorageFile.GetStore (IsolatedStorageScope.User | IsolatedStorageScope.Assembly, null, null);
+
+		/// <summary>
+		/// Sets setting value.
+		/// </summary>
+		/// <param name="key">Key.</param>
+		/// <param name="value">Value.</param>
+		[SuppressMessage ("Microsoft.Usage", "CA2202:Do not dispose objects multiple times", Justification = "ToDo: Fix it later.")]
```
Comment (vguzmanp) [inline VAS.KPI/HockeyPlatformHelperMono.cs]: That's copypasted from their code. And it actually doesn't warn about anything, so Fixed

File: VAS.KPI/HockeyPlatformHelperMono.cs [hunk 5]
```diff
@@ -0,0 +1,320 @@
...
+		{
+			if (isoStore.FileExists (FILE_PREFIX + key)) {
+				isoStore.DeleteFile (FILE_PREFIX + key);
+			}
+		}
+
+
+		#region File access
+
+		// ToDo: Remove warning suppression
```
Comment (vguzmanp) [inline VAS.KPI/HockeyPlatformHelperMono.cs]: Copied from their code

File: VAS.KPI/HockeyPlatformHelperMono.cs [hunk 6]
```diff
@@ -0,0 +1,320 @@
...
+using System.Reflection;
+using System.Security;
+using System.Text;
+using System.Threading.Tasks;
+using Microsoft.HockeyApp;
+
+namespace VAS.KPI
+{
+	/// <summary>
+	/// HockeyPlatformHelper for WPF.
```
Comment (vguzmanp) [inline VAS.KPI/HockeyPlatformHelperMono.cs]: Fixed

File: VAS.KPI/HockeyPlatformHelperMono.cs [hunk 7]
```diff
@@ -0,0 +1,320 @@
...
+		public async Task<IEnumerable<string>> GetFileNamesAsync (string folderName = null, string fileNamePattern = null)
+		{
+			try {
+				string pattern = (folderName ?? "") + Path.DirectorySeparatorChar + fileNamePattern ?? "*";
+				try {
+					// There is a bug in Mono with the IsolatedStorage
+					// https://bugzilla.xamarin.com/show_bug.cgi?id=11771
+					return isoStore.GetFileNames (pattern);
+				} catch (SecurityException) {
+					isoStore.CreateDirectory ((folderName ?? ""));
```
Comment (vguzmanp) [inline VAS.KPI/HockeyPlatformHelperMono.cs]: True, Fixed

File: VAS.KPI/HockeyPlatformHelperMono.cs [hunk 8]
```diff
@@ -0,0 +1,320 @@
...
+                Type.GetType
+                ManagementClass mc = new ManagementClass("Win32_ComputerSystem");
+            //collection to store all management objects
+            ManagementObjectCollection moc = mc.GetInstances();
+            if (moc.Count != 0)
+            {
+                foreach (ManagementObject mo in mc.GetInstances())
+                {
+                 mo["Manufacturer"].ToString()
+                */
```
Comment (vguzmanp) [inline VAS.KPI/HockeyPlatformHelperMono.cs]: Copypasted from their code (even with the deutsch comments 😄

File: VAS.KPI/Services/ApplicationService.cs [hunk 9]
```diff
@@ -0,0 +1,90 @@
+﻿namespace Microsoft.HockeyApp.Services
+{
+	using System;
+	using System.Globalization;
+	using VAS;
```
Comment (vguzmanp) [inline VAS.KPI/Services/ApplicationService.cs]: Yeah... Copypasted too... I don't know why they do that.

File: VAS.KPI/Services/ApplicationService.cs [hunk 10]
```diff
@@ -0,0 +1,90 @@
...
+
+			initialized = true;
+		}
+
+		public bool IsDevelopmentMode ()
+		{
+			// IsDevelopmentMode API is supported only in UWP, for all others return false.
+			return false;
+		}
+
```
Comment (vguzmanp) [inline VAS.KPI/Services/ApplicationService.cs]: Fixed

File: VAS.KPI/Services/StorageService.cs (hunk 1 above)
Comment (vguzmanp) [inline VAS.KPI/Services/StorageService.cs]: I'm not sure about this, as the code is copy-pasted from their own, with a couple modifications

**Pull request merged**